client/src/components/WorkflowInvocationState/WorkflowInvocationOverview.vue

```diff
@@ -108,8 +108,10 @@ const invocationPdfLink = computed<string | null>(() => {
     }
 });
 
-const hasMessages = computed<boolean>(() => {
-    return props.invocation?.messages.length ? true : false;
+const uniqueMessages = computed(() => {
+    const messages = props.invocation?.messages || [];
+    const uniqueMessagesSet = new Set(messages.map((message) => JSON.stringify(message)));
```

Comment:
Ah, why the stringify things ?

Reply:
I see, objects are not hashable. Well, that explains why you're assigning the type.